### PR TITLE
1609: Enable store import koblenz

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

As a store maintainer i want to upload the csv stores for koblenz

### Proposed changes

- move storeConfig to common and use it for koblenz and nuernberg
- add runConfig for project store manager creation for koblenz
- add nuernberg buildConfig as a placeholder for koblenz to get correct categories
- add TODO for getting correct buildConfig in administration. Since nuernberg and koblenz have the same categories we can use the nuernberg build config for now
### Side effects

- none

### Testing

1. Create a store manager for koblenz
2. login as store manager
3. Go to Akzeptanzpartner verwalten
4. Upload stores
Find correct store file here: https://nextcloud.tuerantuer.org/index.php/apps/files/files/6468170
But you can also use same test files as for nuernberg

### Resolved issues

Fixes: #1609

### CODE REVIEW

**Review by ztefanie — CHANGES_REQUESTED** [review]
- Please add an example csv of the store upload to /administration/resources/stores

- Tested and is not working, when using the csv from nextcloud i get this error: Die erforderlichen Spalten sind nicht vorhanden oder nicht in der richtigen Reihenfolge.


Whenever you are sharing a nextcloud link it only opens the landing page for me. I do not have this problem when others are sharing links, i think the correct nextcloud link is:
https://nextcloud.tuerantuer.org/index.php/apps/files/files/6518906?dir=/Projects/Digitale%20Berechtigungskarte/Stadt%20Koblenz/Akzeptanzpartner&openfile=true
Not sure what you are sharing, but it seems not to work.

**Comment by f1sh1918** [comment]
to be honest i have no clue why its not working for you. I attached a working video @ztefanie 


https://github.com/user-attachments/assets/25b4110e-4296-4bea-a4d4-4f0e04df9b4b

I added a sample file to the repository since i don't want to upload the real csv from koblenz

**Comment by ztefanie** [comment]
Tested again, works as expected :tada:

**Review by steffenkleinle — APPROVED** [review]
Tested on firefox, works as expected

File: administration/src/project-configs/storeConfig.ts [hunk 1]
```diff
@@ -8,14 +7,14 @@ import {
...
+import { StoresManagement } from './getProjectConfig'
 import {
   hasMandatoryValue,
   hasValidCategoryId,
   isCoordinate,
   noValidationRequired,
-} from '../helper/storeFieldValidation'
+} from './helper/storeFieldValidation'
 
 export const storeConfig: StoresManagement = {
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply administration/src/project-configs/storeConfig.ts]: i use plural now

File: administration/src/bp-modules/stores/__tests__/AcceptanceStoreEntry.test.ts [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
 import { invalidStoreData, validStoreData } from '../__mock__/mockStoreEntry'
 
-const fields = nuernbergConfig.storeManagement.enabled ? nuernbergConfig.storeManagement.fields : []
 describe('AcceptanceStoreEntry', () => {
-  it('should be validated false store invalid entries', () => {
-    const store = new AcceptingStoreEntry(invalidStoreData, fields)
-    expect(store.isValid()).toBeFalsy()
-  })
+  it.each([{ projectConfig: nuernbergConfig }, { projectConfig: koblenzConfig }])(
+    `should be validated false store invalid entries for $projectConfig.name`,
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply administration/src/bp-modules/stores/__tests__/AcceptanceStoreEntry.test.ts]: i adjusted to sentence so sth clearer

**Comment by f1sh1918** [comment]
> Tested again, works as expected 🎉

can you please recheck and give me an approve if its fine @ztefanie

**Review by ztefanie — CHANGES_REQUESTED** [review]
The columns width is not really good. I do not like the hard coded widths, there are nice approaches to handle this dynamically, see here: https://github.com/palantir/blueprint/issues/1863#issuecomment-559633896

If entering chars (not numbers) or nothing for lat/long i get the wrong error message "Keine passenden Geodaten gefunden! Bitte prüfen sie die Adresse." If the lat/long is invalid, it should either show a message with the info lat/long is wrong or directly get it, if adress is valid.

Special chars should not be allowed in categoryId section, i can input there "10%" and get no error in the ui, but a 500 is thrown when clicking "import stores"

postalcode should only allow numbers

i did not check every field, but the field definitively need checks

**Comment by f1sh1918** [comment]
> Not really happy with only a frontend check, but as this is an endpoint only for logged in users, I think it is oke, but still bad practice.

there is an open issue for that

**Pull request merged**